### ISSUE TITLE
Amazon, and Bestbuy never scrape, return chromium timeout error

### ISSUE THREAD

**Issue by tjpegram** [EricJMarti/inventory-hunter#154]
As the title says, there seems to be an issue that has cropped up with amazon and BestBuy scrapers. They initialize but are no longer able to scrape, chromium just returns a timeout error. 

I believe this is due to a change in these two sites. The program still works for Newegg, B&H and the like.

**Comment by TomTom1991** [comment]
I just tried a little bit and with more cpu allocation it still works. My Ryzen 9 3950X still works fine. Just my raspberrys cpu cant catch up...

**Comment by SandWormZz** [comment]
I'm seeing this too, if I run for long enough it sometimes scrapes. I limited refresh rate / number of urls but I'm still seeing inconsistent scraping from bestbuy/amazon/samsclub/adorama seems like its the website?

**Comment by joshuagascan** [comment]
I'm having the same issue. It seems to scrape at random now sometimes minutes apart. Totally missed the last 3070 FE drop on Best Buy. I reduced my URLs down to just two, one for the 3070 and one for the 3060 TI, but there has been no improvement.

**Comment by cahickm2** [comment]
I'm having the same issue trying to scrape bestbuy with my pi, it initializes but then times out.

ex:
E2021-02-03 13:41:22,493 [bstby_1] caught exception during request: Message: session not created
from timeout: Timed out receiving message from renderer: 600.000
  (Session info: headless chrome=87.0.4280.88)

E2021-02-03 13:41:22,501 [bstby_1] scrape failed

**Comment by Crazyhead90** [comment]
I'm also haveing the same issue with amazon, did anyone find a fix yet?

Other (custom added dutch) sites, seem to work fine, scraping every 2 seconds orso

**Comment by TomTom1991** [comment]
Again, Its not a problem with the scrapers. They work fine. You just need a better processor. The raspberry processer is NOT sufficient. My Ryzen 9 3950X still works fine on amazon & bestbuy.

**Comment by Crazyhead90** [comment]
> 
> 
> Again, Its not a problem with the scrapers. They work fine. You just need a better processor. The raspberry processer is NOT sufficient. My Ryzen 9 3950X still works fine on amazon & bestbuy. But I'm facing the same issues with my raspberry.

Are you sure this is the issue?
I rent the " VPS XL SSD" from contabo and tried to run it there, but i'm getting the same issue there.

I'd expect that the following specs would be MORE than needed:
10 vCPU Cores
60 GB RAM
1.6 TB SSD
1 Gbit/s Port

**Comment by tjpegram** [comment]
So I can verify this, I think, I am not sure what the exact issue is, however I just cloned to my desktop, and ran it under windows, using the default 3070 scraper it starter scraping immediately. However I have seen at least one chromium driver crash. I lengthened the time on my vps config out to 15 seconds, it is currently just chillin after initialization, the first scraper managed to scrape successfully but it has not done anything else since then. 

Will update here as I find out more. I don't have a unix system set up currently with my desktop hardware to test the app there. So I cannot verify if the issue is specific to UNIX/Windows, or if it is a hardware issue

**Comment by tjpegram** [comment]
Ok can anyone with an intel processor verify if the issues are present on intel or not? I've tried a couple of Linux VM's that have identical virtual settings to the VPS I have been having issues with, infact I even neutered the specs some more in the vm's. I was not able to replicate the issue on my local desktop which is AMD Ryzen based. Seems the issue is present on what sounds like all PI/ARM based systems. I suspect the underlying hardware my server host uses is Intel based but I don't know for sure. I cannot think of another hardware issue that would cause this issue, unless is it a clock speed issue, however I cannot test that currently because I have other projects that require my desktop to continue running. I dropped my VM's to as low as 512mb memory allocation, and a single logical thread off my 2700x. I was not able to replicate the issue locally. All instances scraped their links just fine. 
I do not have an intel system to test on, these are the two issues I could see.

**Comment by tjpegram** [comment]
I was able to rob one of my VM's to do some testing. I can confirm it is not an OS Issue. I neutered the vm down to 512mb of ram and a single logical thread of processing power. The scraper still ran fine. But looking at the monitoring for my VPS the processor has been pinged at 100% for the last couple of days, which leads me to believe because of the amount of client side processing built in to amazon and best buy's sites the scrapers for those are crashing out because it takes too long to load because the processor cannot physically process the load. I would suspect that the PI has a similar issue, and since I prevented my VM from doing anything else that core was allowed to do all the processing it needed, and the IPC is fast enough on a single core that it does not matter. 

TLDR: Definitely a hardware issue, wonder if there is a way to prevent some of the javascript processing on amazon and best buy's sites to help this issue

**Comment by Crazyhead90** [comment]
@EricJMarti is your Pi still able to run or are you having the same issue?

If you're able to run it, could you give us all your specs:
Pi#, OS, ram (swap), etc.

**Comment by yardbro** [comment]
Just to add on to what has recently been said here, i'm not entirely convinced there is solely a hardware issue. In my use case scenario I have a VM with 4 vCPUs attached to it and am still experiencing issues with the bestbuy and amazon scrapers only. Up until about a month ago everything related to bestbuy was working perfectly fine but now they are inconsistent at best, sometimes getting hung up for minutes in between scrapes. This is a relatively new Xeon CPU and theres no way a currently supported enterprise CPU can't handle a few scrapers on 4 threads.

**Comment by badasscat2000** [comment]
I've been running inventory hunter on Windows and it throws up the same fatal server errors when opening a container, but then it works fine. I have multiple containers running simultaneously and all work. I just got a Pi 4 today to try to save a few bucks on electricity while running these scrapers (I wanted one for RetroPie anyway) and I'm having the same issues as you guys. But I think the server errors are a red herring since Windows does that too and it still works.

Newegg does still work for me on the Pi; it's mainly Amazon and Best Buy that are the problems, so I do think it's likely something to do with those sites and how the Pi is handling them. My Ryzen 7 3700X is pegged at about 25% most of the time with multiple containers running, but the Pi 4 can't seem to handle even just Amazon or Best Buy by themselves. It does do Newegg ok, though, which strongly suggests to me that it's the javascript on Amazon/BB.

**Comment by tjpegram** [comment]
> I've been running inventory hunter on Windows and it throws up the same fatal server errors when opening a container, but then it works fine. I have multiple containers running simultaneously and all work. I just got a Pi 4 today to try to save a few bucks on electricity while running these scrapers (I wanted one for RetroPie anyway) and I'm having the same issues as you guys. But I think the server errors are a red herring since Windows does that too and it still works.
> 
> Newegg does still work for me on the Pi; it's mainly Amazon and Best Buy that are the problems, so I do think it's likely something to do with those sites and how the Pi is handling them. My Ryzen 7 3700X is pegged at about 25% most of the time with multiple containers running, but the Pi 4 can't seem to handle even just Amazon or Best Buy by themselves. It does do Newegg ok, though, which strongly suggests to me that it's the javascript on Amazon/BB.

Don't go by the total percentage of how much CPU is being used. You want to be looking at the individual core breakdown, most of the containers are going to be likely running on a single core/thread if the core they are running on is pegged out all the time then that is your slowdown where it won't process anymore. Just a thought.
